packages/mgt-components/src/components/mgt-people-picker/mgt-people-picker.ts

```diff
@@ -284,6 +284,17 @@ export class MgtPeoplePicker extends MgtTemplatedComponent {
   })
   public selectionMode: string;
 
+  /**
+   * Get the scopes required for people picker
+   *
+   * @static
+   * @return {*}  {string[]}
+   * @memberof MgtPeoplePicker
+   */
+  public static getScopes(): string[] {
+    return ['user.read.all', 'people.read', 'group.read.all', 'user.readbasic.all'];
```

Comment:
Something that came to mind - sometimes the scopes depend on how the component is used. For example, if a developer only needs to search for users and not groups in the people picker, they don't need the `group.read.all` scope. Similar for person, if a person is not using presence, they don't need the presence scopes.

But not sure how to handle that. Maybe it's a matter of documentation - should we say these scopes are all scopes that "might be needed" based on usage and it's up to the developer to figure out what scopes to really use based on what features they need? Or should we allow some sort of declaration where a developer can declare what features they plan to use and we figure out the scopes?

Reply:
I've been thinking about the same. It's solved very nicely with the person card where you return the list of scopes based on the configuration. But, where person card is being configured globally, all other components are configured independently. So for this to work, we'd either need to have a way to iterate over all components on the page on runtime, examine their configuration and return the necessary scopes or introduce a concept of configuration sets that would allow developers to say which functionality they use and which scopes should be returned, like you mentioned. The current setup is a simple catch-all that works with any functionality that you might use in the component, which is an easy way out for devs, but which could lead to requesting more permissions that actually necessary.